File: backend/app/db/models/enrollment.py
import uuid
from datetime import datetime
from sqlalchemy import Column, Numeric, DateTime, ForeignKey
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship

from app.db.base import Base


class Enrollment(Base):
    """Course enrollment model - tracks user access to courses."""
    __tablename__ = "enrollments"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    user_id = Column(UUID(as_uuid=True), ForeignKey("users.id", ondelete="CASCADE"), nullable=False, index=True)
    course_id = Column(UUID(as_uuid=True), ForeignKey("courses.id", ondelete="CASCADE"), nullable=False, index=True)
    enrolled_at = Column(DateTime, default=datetime.utcnow, nullable=False)
    progress_percent = Column(Numeric(5, 2), default=0.00, nullable=False)

    # Relationships
    user = relationship("User", back_populates="enrollments")
    course = relationship("Course", back_populates="enrollments")

File: frontend/src/types/enrollment.ts
export interface Enrollment {
  id: string;
  user_id: string;
  course_id: string;
  enrolled_at: string;
  progress_percent: number;
}

export interface EnrollmentWithCourse extends Enrollment {
  course_title: string;
  course_slug: string;
  course_thumbnail: string | null;
  course_price: number;
}

export interface CreateEnrollmentRequest {
  course_id: string;
}
